### ISSUE TITLE
Unit test core dumps on macOS

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/arcade#2719]
@filipnavara commented on [Wed Aug 08 2018](https://github.com/dotnet/corefx/issues/31650)

Currently there's a functionality in the test runner code to examine and print out core dump when unit tests crash (with SIGSEGV or similar hard faults). The functionality is restricted to Linux at present. I think it would be beneficial to implement it for macOS as well.

On macOS the core dumps have to be enabled using `ulimit -c unlimited`. Each core dump is placed in `/cores/core.<PID>`. They can be opened using `lldb`, which is already supported in the Linux code.

---

@danmosemsft commented on [Fri Aug 10 2018](https://github.com/dotnet/corefx/issues/31650#issuecomment-412031254)

@ahsonkhan 

---

@wfurt commented on [Wed Sep 19 2018](https://github.com/dotnet/corefx/issues/31650#issuecomment-422658680)

note that we do not build libsosplugin for OSX. So the core and stack-trace will be limited to unmanaged code. (still better then nothing) 

---

@filipnavara commented on [Wed Sep 19 2018](https://github.com/dotnet/corefx/issues/31650#issuecomment-422673847)

That's unfortunate, but managed exception are already reported properly in the test suite anyway. My concern are crashes in `System.Native` and external libraries such as OpenSSL.

---

@wfurt commented on [Wed Sep 19 2018](https://github.com/dotnet/corefx/issues/31650#issuecomment-422889738)

agreed. Saving core file is good thing. This was more a warning for who ever will try to mach Linux capabilities. 

---

@ericstj commented on [Wed May 08 2019](https://github.com/dotnet/corefx/issues/31650#issuecomment-490566292)

@ViktorHofer should this be moved over to Arcade where the test runner lives?

---

@ViktorHofer commented on [Wed May 08 2019](https://github.com/dotnet/corefx/issues/31650#issuecomment-490661575)

There's a related offline discussion about generating dumps for crashes on helix machines, I'm not sure if that's sufficient for this ask. Yes, we should probably move that over as the runner lives in arcade.

**Comment by markwilkie** [comment]
@epananth - does your work address this scenario?

cc/ @natemcmaster

**Comment by epananth** [comment]
Spoke to @ilyas1974  about this, he mentioned that core dumps are only captured for  VSagents but not for Helix. 

I created a new issue to address this for macOS -> https://github.com/dotnet/arcade/issues/2858

@ulisesh is working on testing on prem stuff in staging that's a major change, I will be able to work on this only after that change is merged. 

I am going to be on FR in a week, I can take this up then.

**Comment by epananth** [comment]
** correction @ulisesh will be done with his change this week, so I can work on macOS next week **
